File: app/lib/eltako/api/device.ts
export const targetPosition = "targetPosition"
export const currentPosition = "currentPosition"

export type Info = {
    type: string
    identifier: string
    value: any
}

export type Device = {
    deviceGuid: string
    productGuid: string
    displayName: string
    infos: Info[]
    settings: Info[]
    functions: Info[]
}

File: v2/eltako/device.go
package eltako

type Data struct {
	Type       string      `json:"type"`
	Identifier string      `json:"identifier"`
	Value      interface{} `json:"value"`
}

type Device struct {
	DeviceGuid  string `json:"deviceGuid"`
	ProductGuid string `json:"productGuid"`
	DisplayName string `json:"displayName"`
	Infos       []Data `json:"infos"`
	Settings    []Data `json:"settings"`
	Functions   []Data `json:"functions"`
}
